app/grants/views.py

```diff
@@ -1329,12 +1330,21 @@ def grant_new(request):
             new_grant(grant, profile)
             add_grant_to_active_clrs(grant)
 
+            project_pk = request.POST.get('project_pk', '')
+            if project_pk:
+                HackathonProject.objects.filter(pk=project_pk).update(grant_obj=grant)
+
             return JsonResponse({
                 'success': True,
                 'url': grant.url,
             })
 
-
+    check_profile = None
+    project_data = None
+    project_id = request.GET.get('project_id', None)
+    if project_id is not None:
+        check_profile = request.user.profile if request.user.is_authenticated and hasattr(request.user, 'profile') else None
+        project_data = project(project_id)
```

Comment:
at the beginning of the code is already fetching the user, not need to find it again, also the page is @login_required

Reply:
what need to be checked here if the users "profile.id" is on the project. team_members_profile of someone can just convert projects from others.

Reply:
Done